From a0b47ae9f2e3aa9de5ca30015b56c114fae9fee0 Mon Sep 17 00:00:00 2001
From: kake <kake@build.local>
Date: Mon, 24 Aug 2026 15:57:35 +0000
Subject: [PATCH 06/34] SystemUI: Move network traffic to right side away from
 punch-hole, refine round Android 13 pill tiles and fix customizer glitch

Change-Id: I0f88d90c2f278bbe4a91fdcdfa29fced09084593
---
 packages/SystemUI/res/layout/status_bar.xml   | 16 ----
 .../res/layout/status_bar_toggle_slider.xml   |  8 +-
 packages/SystemUI/res/layout/system_icons.xml | 16 ++++
 packages/SystemUI/res/values/dimens.xml       | 12 +--
 .../systemui/qs/tileimpl/QSTileBaseView.java  | 18 +++--
 .../systemui/qs/tileimpl/QSTileView.java      | 75 +++++++++++--------
 6 files changed, 82 insertions(+), 63 deletions(-)

diff --git a/packages/SystemUI/res/layout/status_bar.xml b/packages/SystemUI/res/layout/status_bar.xml
index 22e82ae8..fe6ead44 100644
--- a/packages/SystemUI/res/layout/status_bar.xml
+++ b/packages/SystemUI/res/layout/status_bar.xml
@@ -109,22 +109,6 @@
             android:clipChildren="false"
             android:gravity="center_horizontal|center_vertical"/>
 
-        <com.android.systemui.statusbar.LineageStatusBarItemHolder
-            android:id="@+id/network_traffic_holder"
-            android:layout_width="wrap_content"
-            android:layout_height="match_parent"
-            android:gravity="start|center_vertical"
-            >
-
-            <org.lineageos.internal.statusbar.NetworkTraffic
-                android:id="@+id/network_traffic"
-                android:layout_width="wrap_content"
-                android:layout_height="match_parent"
-                android:maxLines="2"
-                android:gravity="center_vertical|start" />
-
-        </com.android.systemui.statusbar.LineageStatusBarItemHolder>
-
         <com.android.keyguard.AlphaOptimizedLinearLayout android:id="@+id/system_icon_area"
             android:layout_width="0dp"
             android:layout_height="match_parent"
diff --git a/packages/SystemUI/res/layout/status_bar_toggle_slider.xml b/packages/SystemUI/res/layout/status_bar_toggle_slider.xml
index 942f3dd0..49defd3d 100644
--- a/packages/SystemUI/res/layout/status_bar_toggle_slider.xml
+++ b/packages/SystemUI/res/layout/status_bar_toggle_slider.xml
@@ -36,10 +36,10 @@
         android:layout_centerVertical="true"
         android:layout_alignParentStart="true"
         android:layout_alignParentEnd="true"
-        android:paddingStart="20dp"
-        android:paddingEnd="20dp"
-        android:paddingTop="16dp"
-        android:paddingBottom="16dp"
+        android:paddingStart="0dp"
+        android:paddingEnd="0dp"
+        android:paddingTop="4dp"
+        android:paddingBottom="4dp"
         android:thumb="@drawable/ic_brightness_thumb"
         android:progressDrawable="@drawable/brightness_progress_drawable"
         android:splitTrack="false"
diff --git a/packages/SystemUI/res/layout/system_icons.xml b/packages/SystemUI/res/layout/system_icons.xml
index 3e638599..353f2546 100644
--- a/packages/SystemUI/res/layout/system_icons.xml
+++ b/packages/SystemUI/res/layout/system_icons.xml
@@ -21,6 +21,22 @@
     android:layout_height="match_parent"
     android:gravity="center_vertical">
 
+    <com.android.systemui.statusbar.LineageStatusBarItemHolder
+        android:id="@+id/network_traffic_holder"
+        android:layout_width="wrap_content"
+        android:layout_height="match_parent"
+        android:gravity="center_vertical|end"
+        android:paddingEnd="4dp">
+
+        <org.lineageos.internal.statusbar.NetworkTraffic
+            android:id="@+id/network_traffic"
+            android:layout_width="wrap_content"
+            android:layout_height="match_parent"
+            android:maxLines="2"
+            android:gravity="center_vertical|end" />
+
+    </com.android.systemui.statusbar.LineageStatusBarItemHolder>
+
     <com.android.systemui.statusbar.phone.StatusIconContainer android:id="@+id/statusIcons"
         android:layout_width="0dp"
         android:layout_weight="1"
diff --git a/packages/SystemUI/res/values/dimens.xml b/packages/SystemUI/res/values/dimens.xml
index 60bd8fea..96f02fb1 100644
--- a/packages/SystemUI/res/values/dimens.xml
+++ b/packages/SystemUI/res/values/dimens.xml
@@ -446,10 +446,10 @@
     <!-- The size of the gesture span needed to activate the "pull" notification expansion -->
     <dimen name="pull_span_min">25dp</dimen>
 
-    <dimen name="qs_tile_height">60dp</dimen>
-    <dimen name="qs_tile_layout_margin_side">12dp</dimen>
-    <dimen name="qs_tile_margin_horizontal">6dp</dimen>
-    <dimen name="qs_tile_margin_vertical">6dp</dimen>
+    <dimen name="qs_tile_height">64dp</dimen>
+    <dimen name="qs_tile_layout_margin_side">14dp</dimen>
+    <dimen name="qs_tile_margin_horizontal">8dp</dimen>
+    <dimen name="qs_tile_margin_vertical">8dp</dimen>
     <dimen name="qs_tile_margin_top_bottom">8dp</dimen>
     <dimen name="qs_tile_margin_top_bottom_negative">-8dp</dimen>
     <!-- The height of the qs content, used for calculating the padding on the status bar:
@@ -459,8 +459,8 @@
     <dimen name="qs_total_margin">36dp</dimen>
     <dimen name="qs_tile_margin_top">12dp</dimen>
     <dimen name="qs_tile_background_size">0dp</dimen>
-    <dimen name="qs_quick_tile_size">40dp</dimen>
-    <dimen name="qs_quick_tile_padding">6dp</dimen>
+    <dimen name="qs_quick_tile_size">48dp</dimen>
+    <dimen name="qs_quick_tile_padding">10dp</dimen>
     <dimen name="qs_tile_icon_bg_margin">0dp</dimen>
     <dimen name="qs_tile_side_label_padding">0dp</dimen>
     <dimen name="qs_tile_icon_size">22dp</dimen>
diff --git a/packages/SystemUI/src/com/android/systemui/qs/tileimpl/QSTileBaseView.java b/packages/SystemUI/src/com/android/systemui/qs/tileimpl/QSTileBaseView.java
index 0ab96321..0d3cd922 100644
--- a/packages/SystemUI/src/com/android/systemui/qs/tileimpl/QSTileBaseView.java
+++ b/packages/SystemUI/src/com/android/systemui/qs/tileimpl/QSTileBaseView.java
@@ -120,9 +120,13 @@ public class QSTileBaseView extends com.android.systemui.plugins.qs.QSTileView {
 
     protected Drawable newTileBackground() {
         mPillBackground = new GradientDrawable();
-        mPillBackground.setShape(GradientDrawable.RECTANGLE);
-        float radius = getContext().getResources().getDisplayMetrics().density * 16f;
-        mPillBackground.setCornerRadius(radius);
+        float radius = getContext().getResources().getDisplayMetrics().density * 24f;
+        if (mCollapsedView) {
+            mPillBackground.setShape(GradientDrawable.OVAL);
+        } else {
+            mPillBackground.setShape(GradientDrawable.RECTANGLE);
+            mPillBackground.setCornerRadius(radius);
+        }
         mPillBackground.setColor(0xFF282C37);
 
         int[] attrs = new int[]{android.R.attr.colorControlHighlight};
@@ -131,8 +135,12 @@ public class QSTileBaseView extends com.android.systemui.plugins.qs.QSTileView {
         ta.recycle();
 
         GradientDrawable mask = new GradientDrawable();
-        mask.setShape(GradientDrawable.RECTANGLE);
-        mask.setCornerRadius(radius);
+        if (mCollapsedView) {
+            mask.setShape(GradientDrawable.OVAL);
+        } else {
+            mask.setShape(GradientDrawable.RECTANGLE);
+            mask.setCornerRadius(radius);
+        }
         mask.setColor(Color.WHITE);
 
         return new RippleDrawable(ColorStateList.valueOf(rippleColor), mPillBackground, mask);
diff --git a/packages/SystemUI/src/com/android/systemui/qs/tileimpl/QSTileView.java b/packages/SystemUI/src/com/android/systemui/qs/tileimpl/QSTileView.java
index 11b9f4cf..22fc7954 100644
--- a/packages/SystemUI/src/com/android/systemui/qs/tileimpl/QSTileView.java
+++ b/packages/SystemUI/src/com/android/systemui/qs/tileimpl/QSTileView.java
@@ -62,15 +62,18 @@ public class QSTileView extends QSTileBaseView {
 
         setClickable(true);
         setId(View.generateViewId());
-        setOrientation(HORIZONTAL);
-        setGravity(Gravity.CENTER_VERTICAL | Gravity.START);
-        int padH = (int) (getContext().getResources().getDisplayMetrics().density * 12f);
-        int padV = (int) (getContext().getResources().getDisplayMetrics().density * 6f);
-        setPaddingRelative(padH, padV, padH, padV);
-        createLabel();
+        if (!collapsedView) {
+            setOrientation(HORIZONTAL);
+            setGravity(Gravity.CENTER_VERTICAL | Gravity.START);
+            int padH = (int) (getContext().getResources().getDisplayMetrics().density * 14f);
+            int padV = (int) (getContext().getResources().getDisplayMetrics().density * 8f);
+            setPaddingRelative(padH, padV, padH, padV);
+            createLabel();
+        } else {
+            setOrientation(VERTICAL);
+            setGravity(Gravity.CENTER);
+        }
         mColorLabelDefault = Utils.getColorAttr(getContext(), android.R.attr.textColorPrimary);
-        // The text color for unavailable tiles is textColorSecondary, same as secondaryLabel for
-        // contrast purposes
         mColorLabelUnavailable = Utils.getColorAttr(getContext(),
                 android.R.attr.textColorSecondary);
     }
@@ -82,13 +85,13 @@ public class QSTileView extends QSTileBaseView {
     @Override
     protected void onConfigurationChanged(Configuration newConfig) {
         super.onConfigurationChanged(newConfig);
-        FontSizeUtils.updateFontSize(mLabel, R.dimen.qs_tile_text_size);
-        FontSizeUtils.updateFontSize(mSecondLine, R.dimen.qs_tile_text_size);
+        if (mLabel != null) FontSizeUtils.updateFontSize(mLabel, R.dimen.qs_tile_text_size);
+        if (mSecondLine != null) FontSizeUtils.updateFontSize(mSecondLine, R.dimen.qs_tile_text_size);
     }
 
     @Override
     public int getDetailY() {
-        return getTop() + mLabelContainer.getTop() + mLabelContainer.getHeight() / 2;
+        return mLabelContainer != null ? (getTop() + mLabelContainer.getTop() + mLabelContainer.getHeight() / 2) : getTop() + getHeight() / 2;
     }
 
     protected void createLabel() {
@@ -103,6 +106,9 @@ public class QSTileView extends QSTileBaseView {
         mExpandSpace = mLabelContainer.findViewById(R.id.expand_space);
         mSecondLine = mLabelContainer.findViewById(R.id.app_label);
 
+        if (mExpandIndicator != null) mExpandIndicator.setVisibility(View.GONE);
+        if (mExpandSpace != null) mExpandSpace.setVisibility(View.GONE);
+
         LinearLayout.LayoutParams lp = new LinearLayout.LayoutParams(0, ViewGroup.LayoutParams.WRAP_CONTENT, 1.0f);
         lp.gravity = Gravity.CENTER_VERTICAL;
         lp.setMarginStart((int) (getContext().getResources().getDisplayMetrics().density * 8f));
@@ -113,10 +119,8 @@ public class QSTileView extends QSTileBaseView {
     protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
         super.onMeasure(widthMeasureSpec, heightMeasureSpec);
 
-        // Remeasure view if the primary label requires more then 2 lines or the secondary label
-        // text will be cut off.
-        if (mLabel.getLineCount() > MAX_LABEL_LINES || !TextUtils.isEmpty(mSecondLine.getText())
-                        && mSecondLine.getLineHeight() > mSecondLine.getHeight()) {
+        if (mLabel != null && (mLabel.getLineCount() > MAX_LABEL_LINES || !TextUtils.isEmpty(mSecondLine.getText())
+                        && mSecondLine.getLineHeight() > mSecondLine.getHeight())) {
             mLabel.setSingleLine();
             super.onMeasure(widthMeasureSpec, heightMeasureSpec);
         }
@@ -125,7 +129,7 @@ public class QSTileView extends QSTileBaseView {
     @Override
     protected void handleStateChanged(QSTile.State state) {
         super.handleStateChanged(state);
-        if (!Objects.equals(mLabel.getText(), state.label) || mState != state.state) {
+        if (mLabel != null && (!Objects.equals(mLabel.getText(), state.label) || mState != state.state)) {
             if (state.state == Tile.STATE_ACTIVE) {
                 mLabel.setTextColor(0xFF000000);
             } else if (state.state == Tile.STATE_UNAVAILABLE) {
@@ -136,7 +140,7 @@ public class QSTileView extends QSTileBaseView {
             mState = state.state;
             mLabel.setText(state.label);
         }
-        if (!Objects.equals(mSecondLine.getText(), state.secondaryLabel) || mState != state.state) {
+        if (mSecondLine != null && (!Objects.equals(mSecondLine.getText(), state.secondaryLabel) || mState != state.state)) {
             if (state.state == Tile.STATE_ACTIVE) {
                 mSecondLine.setTextColor(0x99000000);
             } else if (state.state == Tile.STATE_UNAVAILABLE) {
@@ -148,27 +152,34 @@ public class QSTileView extends QSTileBaseView {
             mSecondLine.setVisibility(TextUtils.isEmpty(state.secondaryLabel) ? View.GONE
                     : View.VISIBLE);
         }
-        boolean dualTarget = DUAL_TARGET_ALLOWED && state.dualTarget;
-        mExpandIndicator.setVisibility(dualTarget ? View.VISIBLE : View.GONE);
-        mExpandSpace.setVisibility(dualTarget ? View.VISIBLE : View.GONE);
-        mLabelContainer.setContentDescription(dualTarget ? state.dualLabelContentDescription
-                : null);
-        if (dualTarget != mLabelContainer.isClickable()) {
-            mLabelContainer.setClickable(dualTarget);
-            mLabelContainer.setLongClickable(dualTarget);
-            mLabelContainer.setBackground(dualTarget ? newTileBackground() : null);
+        if (mExpandIndicator != null) {
+            mExpandIndicator.setVisibility(View.GONE);
+        }
+        if (mExpandSpace != null) {
+            mExpandSpace.setVisibility(View.GONE);
+        }
+        if (mLabelContainer != null) {
+            mLabelContainer.setClickable(false);
+            mLabelContainer.setLongClickable(false);
+            mLabelContainer.setBackground(null);
+        }
+        if (mLabel != null) {
+            mLabel.setEnabled(!state.disabledByPolicy);
+        }
+        if (mPadLock != null) {
+            mPadLock.setVisibility(state.disabledByPolicy ? View.VISIBLE : View.GONE);
         }
-        mLabel.setEnabled(!state.disabledByPolicy);
-        mPadLock.setVisibility(state.disabledByPolicy ? View.VISIBLE : View.GONE);
     }
 
     @Override
     public void init(OnClickListener click, OnClickListener secondaryClick,
             OnLongClickListener longClick) {
         super.init(click, secondaryClick, longClick);
-        mLabelContainer.setOnClickListener(secondaryClick);
-        mLabelContainer.setOnLongClickListener(longClick);
-        mLabelContainer.setClickable(false);
-        mLabelContainer.setLongClickable(false);
+        if (mLabelContainer != null) {
+            mLabelContainer.setOnClickListener(secondaryClick);
+            mLabelContainer.setOnLongClickListener(longClick);
+            mLabelContainer.setClickable(false);
+            mLabelContainer.setLongClickable(false);
+        }
     }
 }
-- 
2.43.0

